### DISCUSSION TITLE
Should we completely follow ClickHouse query syntax?

### DISCUSSION THREAD

**Discussion by TennyZhuang** [databendlabs/databend#4742] [Q&A] [answered]
ClickHouse use a `LIMIT n[, m]` syntax, rather than `LIMIT n OFFSET m` in mysql. and sqlparse-rs also follows `LIMIT OFFSET` syntax. Should we modify it in https://github.com/datafuse-extras/sqlparser-rs ?

**Comment by TennyZhuang** [comment]
BTW

The ClickHouse query syntax is ugly 😅

**Comment by bohutang** [comment, marked as answer]
Hello @TennyZhuang 
Datafuse mainly follow the MySQL syntax ：）
But the plan_parser.rs currently not clear enough and needs to be improved.

**Comment by sundy-li** [comment]
Seems mysql supports both for limit offset syntax.

https://www.db-fiddle.com/f/6rRk95Uk4V2oQnXMwxQa3f/0
